api/PclusterApiHandler.py

```diff
@@ -271,6 +271,57 @@ def submit_job():
     return resp if type(resp) == tuple else {"success": "true"}
 
 
+def translate_job(request_body, user):
+    # format job parameters so that they are accepted by Slurm API
+    job_properties_dict = {
+        "job-name": "name",
+        "chdir": "current_working_directory"
+    }
+
+    translated_job = {
+        "job": {
+           "environment": {
+                "PATH": "/bin:/usr/bin/:/usr/local/bin/"
+            } 
+        }
+    }
+
+    request_body["chdir"] = f"/home/{user}"
+    for key in request_body:
+        if key == "command":
+            translated_job["script"] = request_body[key]
+        else:
+            translated_job["job"][job_properties_dict.get(key, key)] = request_body[key]
+
+    return translated_job
+
+
+def submit_job_script():
+    body = request.json
+    cluster_name = request.args.get("cluster_name")
+    instance_id = request.args.get("instance_id")
+    region = request.args.get("region")
+    user = request.args.get("user", "ec2-user")
+    body_data = json.dumps(translate_job(body, user)) if body else None
+
+    ec2 = boto3.resource("ec2", region_name=region)
+    instance = ec2.Instance(instance_id)
+    ip = instance.public_dns_name
+
+    client = boto3.client("secretsmanager")
+    jwt_token = client.get_secret_value(SecretId="slurm_token_"+cluster_name)["SecretString"]
+
+    url = "https://"+ip+"/slurm/v0.0.36/job/submit"
+    headers = {
+        "Content-Type": "application/json",
+        "X-SLURM-USER-NAME": user,
+        "X-SLURM-USER-TOKEN": jwt_token
+    }
+
+    resp = requests.post(url=url, data=body_data, headers=headers, verify=False)
+    return resp if type(resp) == tuple else {"success": "true"}
+
```

Comment:
issue(blocking): This handler does not have a test. In order to make it easier to test, you could extract the actual business logic into a separate function and test that:

```python
def actual_business_logic(body, user, token, ip):
 # generate body_data
 # request.post
 # generate response
```

You can then test the request is performed correctly (url, headers and body), and check the response is the one you expect (either the actual response or the fallback one)